### PR TITLE
Added named groups babel transform to rn babel preset

### PR DESCRIPTION
**Pull request by danilobuerger**
## Summary

Hermes currently does not support regex named groups. Whenever a library is included that uses named groups the rn build with hermes will fail (as it doesn't compile). This is already the case when using jest 27 as this includes a dependency (ansi-styles) that uses named-groups which is transitively included via pretty-format in HMRClient.

## Test plan

no test plan

### CODE REVIEW

File: packages/metro-react-native-babel-preset/src/configs/main.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ const defaultPlugins = [
   [require('@babel/plugin-syntax-dynamic-import')],
   [require('@babel/plugin-syntax-export-default-from')],
   ...passthroughSyntaxPlugins,
+  [require('@babel/plugin-transform-named-capturing-groups-regex')],
```
Comment (newyankeecodeshop) [inline packages/metro-react-native-babel-preset/src/configs/main.js:34]: I think JavaScriptCore for iOS 11+ has this feature - maybe this plugin should be added in the "extraPlugins" section only `if (isHermes)`?
The Android JavaScriptCore has functionality of about the iOS 12 release, so I expect it also has this feature.

Comment (danilobuerger) [reply packages/metro-react-native-babel-preset/src/configs/main.js]: Thanks for the review @newyankeecodeshop . I updated the PR with your recommendation.

**Comment by danilobuerger** [comment]
@robhogan anything I can do about the internal test fails?

**Comment by robhogan** [comment]
Thanks but I've taken care of it - just an internal lockfile needed syncing.

**Pull request closed without merging**